src/NLog/Targets/FileTarget.cs

```diff
@@ -2203,12 +2219,6 @@ private void ConditionalFlushOpenFileAppenders()
         /// <param name="initializedNewFile">File has just been opened.</param>
         private void WriteToFile(string fileName, ArraySegment<byte> bytes, bool initializedNewFile)
         {
-            if (ReplaceFileContentsOnEachWrite)
```

Comment:
is this correct? This a behavior change? (WriteToFile calls this one, but ok, replace will be strange with a footer)

Reply:
I think this is more correct. Since as you have also noticed, that WriteToFile is reused when writing Footer.

Reply:
maybe create an issue and resolve it with this PR? It's a bit hidden now.

Reply:
I'm ok it is hidden. Think `ReplaceFileContentsOnEachWrite` is something very occult and special.

Reply:
Issue #3889

Reply:
I'm wrong here? There is no bug? See unit test ReplaceFileContentsOnEachWriteTest

Reply:
There is a bug when using `ReplaceFileContentsOnEachWrite` together with  `Footer`. But both features are rather exotic.

Reply:
but that case unit tested in ReplaceFileContentsOnEachWriteTest?

Reply:
Again exotic features, so you have to read the code to get the documentation. Mind I didn't create `ReplaceFileContentsOnEachWrite` :). But footer is written by FinalizeFile, so if the ReplaceFileContentsOnEachWrite moved the FileTarget into a state where it performed Finalize then it would fail. I guess `WriteFooter`-method should return early when `ReplaceFileContentsOnEachWrite = true`.